File: backend/app/schemas/image_stock.py
"""
配图功能相关 Schema
支持图库搜索（Unsplash/Pexels）和 AI 生成
"""
from typing import List, Optional
from pydantic import BaseModel, Field
from enum import Enum


class ImageSource(str, Enum):
    """图片来源"""
    UNSPLASH = "unsplash"
    PEXELS = "pexels"
    AI_GENERATED = "ai_generated"


class ImageOrientation(str, Enum):
    """图片方向"""
    LANDSCAPE = "landscape"  # 横向
    PORTRAIT = "portrait"    # 纵向
    SQUARE = "square"        # 正方形


class ImageSearchRequest(BaseModel):
    """图库搜索请求"""
    query: str = Field(..., description="搜索关键词", min_length=1, max_length=100)
    source: Optional[ImageSource] = Field(None, description="指定图库来源，不指定则搜索全部")
    orientation: Optional[ImageOrientation] = Field(None, description="图片方向")
    page: int = Field(1, description="页码", ge=1)
    per_page: int = Field(20, description="每页数量", ge=1, le=30)
    color: Optional[str] = Field(None, description="主色调（仅 Unsplash 支持）")


class ImageItem(BaseModel):
    """图片项"""
    id: str = Field(..., description="图片ID")
    source: ImageSource = Field(..., description="图片来源")
    url: str = Field(..., description="图片URL")
    thumb_url: str = Field(..., description="缩略图URL")
    width: int = Field(..., description="图片宽度")
    height: int = Field(..., description="图片高度")
    alt: Optional[str] = Field(None, description="图片描述")
    photographer: Optional[str] = Field(None, description="摄影师")
    photographer_url: Optional[str] = Field(None, description="摄影师主页")
    download_url: Optional[str] = Field(None, description="下载链接")
    color: Optional[str] = Field(None, description="主色调")


class ImageSearchResponse(BaseModel):
    """图库搜索响应"""
    query: str = Field(..., description="搜索关键词")
    total: int = Field(..., description="总数量")
    page: int = Field(..., description="当前页码")
    per_page: int = Field(..., description="每页数量")
    images: List[ImageItem] = Field(default_factory=list, description="图片列表")


class KeywordSuggestRequest(BaseModel):
    """关键词建议请求"""
    content: str = Field(..., description="文章内容或主题", min_length=2, max_length=1000)
    count: int = Field(5, description="建议数量", ge=1, le=10)


class KeywordSuggestResponse(BaseModel):
    """关键词建议响应"""
    keywords: List[str] = Field(..., description="建议的搜索关键词")
    keywords_en: List[str] = Field(..., description="英文关键词（图库搜索更准确）")

File: frontend/src/api/imageStock.ts
/**
 * 图库搜索 API
 */
import request from './request'

// 图片来源枚举
export enum ImageSource {
  UNSPLASH = 'unsplash',
  PEXELS = 'pexels',
  AI_GENERATED = 'ai_generated',
}

// 图片方向枚举
export enum ImageOrientation {
  LANDSCAPE = 'landscape',   // 横向
  PORTRAIT = 'portrait',     // 纵向
  SQUARE = 'square',         // 正方形
}

// 图库搜索请求
export interface ImageSearchRequest {
  query: string              // 搜索关键词
  source?: ImageSource       // 图库来源
  orientation?: ImageOrientation // 图片方向
  page?: number              // 页码
  per_page?: number          // 每页数量
  color?: string             // 主色调
}

// 图片项
export interface ImageItem {
  id: string                 // 图片ID
  source: ImageSource        // 图片来源
  url: string                // 图片URL
  thumb_url: string          // 缩略图URL
  width: number              // 图片宽度
  height: number             // 图片高度
  alt?: string               // 图片描述
  photographer?: string      // 摄影师
  photographer_url?: string  // 摄影师主页
  download_url?: string      // 下载链接
  color?: string             // 主色调
}

// 图库搜索响应
export interface ImageSearchResponse {
  query: string              // 搜索关键词
  total: number              // 总数量
  page: number               // 当前页码
  per_page: number           // 每页数量
  images: ImageItem[]        // 图片列表
}

// 关键词建议请求
export interface KeywordSuggestRequest {
  content: string            // 文章内容或主题
  count?: number             // 建议数量
}

// 关键词建议响应
export interface KeywordSuggestResponse {
  keywords: string[]         // 中文关键词
  keywords_en: string[]      // 英文关键词
}

// 图库源状态
export interface SourceStatus {
  configured: boolean
  name: string
  rate_limit: string
}

/**
 * 获取图库源配置状态
 */
export function getSourcesStatus() {
  return request.get<Record<string, SourceStatus>>('/v1/image-stock/sources')
}

/**
 * 搜索图库
 * @param params 搜索参数
 */
export function searchImages(params: ImageSearchRequest) {
  return request.get<ImageSearchResponse>('/v1/image-stock/search', { params })
}

/**
 * 搜索图库 (POST)
 * @param data 搜索参数
 */
export function searchImagesPost(data: ImageSearchRequest) {
  return request.post<ImageSearchResponse>('/v1/image-stock/search', data)
}

/**
 * 根据内容建议搜索关键词
 * @param data 关键词建议请求
 */
export function suggestKeywords(data: KeywordSuggestRequest) {
  return request.post<KeywordSuggestResponse>('/v1/image-stock/suggest-keywords', data)
}

// 图库来源标签
export const SOURCE_LABELS: Record<ImageSource, string> = {
  [ImageSource.UNSPLASH]: 'Unsplash',
  [ImageSource.PEXELS]: 'Pexels',
  [ImageSource.AI_GENERATED]: 'AI 生成',
}

// 图片方向标签
export const ORIENTATION_LABELS: Record<ImageOrientation, string> = {
  [ImageOrientation.LANDSCAPE]: '横向',
  [ImageOrientation.PORTRAIT]: '纵向',
  [ImageOrientation.SQUARE]: '正方形',
}
